File: backend/app/Models/Bid.php
<?php

namespace App\Models;

use App\Traits\HasAuditLog;
use App\Traits\HasTenantScope;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Bid extends Model
{
    use HasFactory, HasTenantScope, HasUuids, HasAuditLog;

    protected $fillable = [
        'tenant_id',
        'tender_id',
        'supplier_id',
        'total_amount',
        'currency',
        'delivery_days',
        'technical_notes',
        'status',
        'submitted_at',
        'weighted_score',
    ];

    protected $casts = [
        'total_amount'   => 'string',
        'weighted_score' => 'string',
        'submitted_at'   => 'datetime',
    ];

    // -------------------------------------------------------------------------
    // Relationships
    // -------------------------------------------------------------------------

    public function tenant(): BelongsTo
    {
        return $this->belongsTo(Tenant::class);
    }

    public function tender(): BelongsTo
    {
        return $this->belongsTo(Tender::class);
    }

    public function supplier(): BelongsTo
    {
        return $this->belongsTo(Supplier::class);
    }

    public function documents(): HasMany
    {
        return $this->hasMany(BidDocument::class);
    }

    public function evaluations(): HasMany
    {
        return $this->hasMany(BidEvaluation::class);
    }

    public function purchaseOrder(): HasOne
    {
        return $this->hasOne(PurchaseOrder::class);
    }
}

File: mobile/lib/features/bids/domain/entities/bid.dart
import 'package:equatable/equatable.dart';

class Bid extends Equatable {
  final String id;
  final String tenderId;
  final String supplierId;
  final double totalAmount;
  final String currency;
  final int deliveryDays;
  final String? technicalNotes;
  final String status;
  final DateTime? submittedAt;

  const Bid({
    required this.id,
    required this.tenderId,
    required this.supplierId,
    required this.totalAmount,
    required this.currency,
    required this.deliveryDays,
    this.technicalNotes,
    required this.status,
    this.submittedAt,
  });

  @override
  List<Object?> get props => [
        id,
        tenderId,
        supplierId,
        totalAmount,
        currency,
        deliveryDays,
        technicalNotes,
        status,
        submittedAt,
      ];
}
